elements/pf-chip/pf-chip.ts

```diff
@@ -0,0 +1,118 @@
+import { LitElement, html, type PropertyValueMap } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+import { property } from 'lit/decorators/property.js';
+import { query } from 'lit/decorators/query.js';
+import '@patternfly/elements/pf-button/pf-button.js';
+
+import styles from './pf-chip.css';
+
+export class ChipReadyEvent extends Event {
+  constructor() {
+    super('ready', { bubbles: true });
+  }
+}
+export class ChipRemoveEvent extends Event {
+  constructor() {
+    super('remove', { bubbles: true });
+  }
+}
+
+/**
+ * A **chip** is used to communicate a value or a set of attribute-value pairs within workflows that involve filtering a set of objects.
+ *
+ * @fires { ChipReadyEvent } ready - Fires when chip is ready
+ * @fires { ChipRemoveEvent } remove - Fires when chip is removed
+ * @fires { Event } click - when close button is clicked
+ *
+ * @slot
+ *      chip text
+ *
+ * @csspart text - span container for chip text
+ */
+@customElement('pf-chip')
+export class PfChip extends LitElement {
+  static readonly styles = [styles];
+  static override readonly shadowRootOptions: ShadowRootInit = { ...LitElement.shadowRootOptions, delegatesFocus: true };
+
+  /**
+   * Accessible label for close button
+   */
+  @property({ attribute: 'accessible-close-label', type: String }) accessibleCloseLabel = 'Close';
+
+  /**
+   * Flag indicating if chip is an overflow chip
+   */
+  @property({ reflect: true, attribute: 'overflow-chip', type: Boolean }) overflowChip = false;
```

Comment:
should this be a public api? does it ever make sense to have an "overflow chip" which isn't automatically generated?